File: src/main/java/org/taktik/connector/technical/validator/impl/SAMLSessionValidator.java
package org.taktik.connector.technical.validator.impl;

import org.taktik.connector.technical.exception.SessionManagementException;
import org.taktik.connector.technical.exception.SessionManagementExceptionValues;
import org.taktik.connector.technical.service.sts.security.SAMLToken;
import org.taktik.connector.technical.service.sts.utils.SAMLHelper;
import org.taktik.connector.technical.session.Session;
import org.taktik.connector.technical.validator.SessionValidator;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class SAMLSessionValidator implements SessionValidator {
   private static final Logger LOG = LoggerFactory.getLogger(SAMLSessionValidator.class);

   public boolean validateSession() throws SessionManagementException {
      if (!Session.getInstance().hasValidSession()) {
         LOG.error("No valid session found");
         throw new SessionManagementException(SessionManagementExceptionValues.ERROR_NOSESSION);
      } else {
         return true;
      }
   }

   public boolean validateToken(SAMLToken samlToken) throws SessionManagementException {
      if (samlToken != null && samlToken.getAssertion() != null) {
         return SAMLHelper.getNotOnOrAfterCondition(samlToken.getAssertion()).isAfterNow();
      } else {
         LOG.error("No valid samlToken");
         throw new SessionManagementException(SessionManagementExceptionValues.ERROR_NOTOKEN);
      }
   }
}

File: src/main/kotlin/org/taktik/connector/technical/validator/impl/SAMLSessionValidator.kt
package org.taktik.connector.technical.validator.impl

import org.taktik.connector.technical.exception.SessionManagementException
import org.taktik.connector.technical.exception.SessionManagementExceptionValues
import org.taktik.connector.technical.service.sts.security.SAMLToken
import org.taktik.connector.technical.service.sts.utils.SAMLHelper
import org.taktik.connector.technical.session.Session
import org.taktik.connector.technical.validator.SessionValidator
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class SAMLSessionValidator : SessionValidator {

    @Throws(SessionManagementException::class)
    override fun validateSession(): Boolean {
        if (!Session.getInstance().hasValidSession()) {
            LOG.error("No valid session found")
            throw SessionManagementException(SessionManagementExceptionValues.ERROR_NOSESSION)
        } else {
            return true
        }
    }

    @Throws(SessionManagementException::class)
    override fun validateToken(samlToken: SAMLToken?): Boolean {
        if (samlToken != null && samlToken.assertion != null) {
            return SAMLHelper.getNotOnOrAfterCondition(samlToken.assertion)!!.isAfterNow
        } else {
            LOG.error("No valid samlToken")
            throw SessionManagementException(SessionManagementExceptionValues.ERROR_NOTOKEN)
        }
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(SAMLSessionValidator::class.java)
    }
}
